### PR TITLE
refactor vision models

### PR DESCRIPTION
**Pull request by szha**
refactored existing vision models and added VGG example using nn interface.

### CODE REVIEW

**Comment by mli** [comment]
given that there is already a resnet, i don't think an additional vgg will help too much. instead, i suggest to put vgg and others in a model package, so users can get it through `from foo.model import vgg`

**Comment by szha** [comment]
@mli will update following similar structure in image-classification. This was added because the fast-neural-style transfer example depended on VGG.

File: python/mxnet/foo/model/vision.py [hunk 1]
```diff
@@ -1,37 +1,14 @@
 from __future__ import division
 
-import argparse, time
```
Comment (piiswrong) [inline python/mxnet/foo/model/vision.py]: let's put this in example for now and move to foo.model after the interface is stable.

Comment (szha) [inline python/mxnet/foo/model/vision.py]: OK. Then we just need to keep in mind that we should maintain this for future interface changes.

**Comment by szha** [comment]
Moved model/vision.py back to example as vision_model.py for now. Also fixed lint along the way. Let me do more testing before merging.

File: example/autograd/vision_model.py [hunk 2]
```diff
@@ -269,105 +250,122 @@ def hybrid_forward(self, F, x):
...
-                152: ('bottle_neck', [3, 8, 36, 3], [64, 256, 512, 1024, 2048]) }
+
+class VGG(nn.HybridLayer):
+    def __init__(self, classes, layers, filters, batch_norm, **kwargs):
+        super(VGG, self).__init__(**kwargs)
+        with self.name_scope():
+            assert len(layers) == len(filters)
+            self.features = self._make_features(layers, filters, batch_norm)
+            self.classifier = nn.HSequential()
+            self.classifier.add(nn.Dense(4096, activation='relu',
```
Comment (piiswrong) [inline example/autograd/vision_model.py]: with self.classifier.name_scope

File: example/autograd/vision_model.py [hunk 3]
```diff
@@ -269,105 +250,122 @@ def hybrid_forward(self, F, x):
...
+                                         kernel_initializer=initializer.Normal(),
+                                         bias_initializer='zeros'))
+            self.classifier.add(nn.Dropout(rate=0.5))
+            self.classifier.add(nn.Dense(classes,
+                                         kernel_initializer=initializer.Normal(),
+                                         bias_initializer='zeros'))
+
+    def _make_features(self, layers, filters, batch_norm):
+        featurizer = nn.HSequential()
+        for i, num in enumerate(layers):
```
Comment (piiswrong) [inline example/autograd/vision_model.py]: same

**Comment by szha** [comment]
closing now to update to the new interface. will resend

**Pull request closed without merging**